src/tree/common_row_partitioner.h

```diff
@@ -194,7 +207,12 @@ class CommonRowPartitioner {
     std::vector<int32_t> split_conditions;
     if (column_matrix.IsInitialized()) {
       split_conditions.resize(n_nodes);
-      FindSplitConditions(nodes, *p_tree, gmat, &split_conditions);
+      if (is_secure_) {
+        // in secure mode, the split index is kept instead of the split value
+        FindSplitConditions(nodes, *p_tree, gmat, &split_conditions, true);
```

Comment:
The condition looks exactly the same as: `FindSplitConditions(nodes, *p_tree, gmat, &split_conditions, is_secure_);`

Reply:
thanks! will check

Reply:
after the above change, this is no longer needed